consensus/src/simplex/mocks/application.rs

```diff
@@ -242,18 +242,23 @@ impl<E: Clock + RngCore, H: Hasher> Application<E, H> {
         digest
     }
 
-    async fn verify(&mut self, context: Context, payload: Digest, mut contents: Bytes) -> bool {
+    async fn verify(
+        &mut self,
+        context: Context,
+        payload: DigestBytes,
+        mut contents: Bytes,
+    ) -> bool {
         // Simulate the verify latency
         let duration = self.verify_latency.sample(&mut self.runtime);
         self.runtime
             .sleep(Duration::from_millis(duration as u64))
             .await;
 
         // Verify parent exists and we are at the correct height
-        if !H::validate(&context.parent.1) {
+        if !H::validate(&H::from(&context.parent.1)) {
             self.panic("invalid parent digest length");
         }
-        if !H::validate(&payload) {
+        if !H::validate(&H::from(&payload)) {
```

Comment:
I suggest we remove `H::from/validate` and instead require `H::Digest` to implement `TryFrom<>`. I spiked on this a bit in my branch.

Reply:
Today, I did most of the changes regarding what you propose in the spike.
Do you agree Validator<[u8]> trait looks a bit redondant if TryFrom<[u8]> is implemented?

Reply:
Yeah, I think we can entirely remove `Validator` and use the built-in `TryFrom`.